src/detection_linux.cpp

```diff
@@ -244,6 +241,9 @@ static void cbWork(uv_work_t *req) {
 }
 
 static void cbAfter(uv_work_t *req, int status) {
+	udev_monitor_unref(mon);
+	udev_unref(udev);
```

Comment:
Why can't we just move these above `uv_mutex_destroy` in `Stop`? It seems like it would have the same effect.

I assume there is nuance between `cbAfter` vs `cbTerminate` that we want this difference. Why don't we want to `udev_unref(udev);` in `cbTerminate`?

Reply:
from some googling, I think the flow is that after cbWork(running on the threadpool) has finished, cbAfter(run on the event loop) will be called.
cbTerminate will be called (I assume on the event loop) when the program wishes to terminate.

It sounds to me like cbTerminate wants to simply set isRunning to false, and let cbAfter do all the cleanup. 
I think the problem is that the poll loop is in the middle of a poll at the point where `udev_monitor_unref(mon);` is called, causing undefined behaviour

A follow up question is why does cbAfter/ cbWork do the cleanup instead of doing it at the end of cbWork after the loop has stopped?

Reply:
> It sounds to me like `cbTerminate` wants to simply set `isRunning` to `false`, and let `cbAfter` do all the cleanup. 
I think the problem is that the poll loop is in the middle of a poll at the point where `udev_monitor_unref(mon);` is called, causing undefined behaviour

@Julusian This sounds reasonable 👍

> A follow up question is why does `cbAfter`/`cbTerminate` do the cleanup instead of doing it at the end of `cbWork` after the loop has stopped?

This also seems reasonable as long as those clean-up functions can be called there 🤷 

Appreciate the context and confidence here ❤️